contents/staticAnalysis/ESLint.md

```diff
@@ -0,0 +1,109 @@
+# ESLint
+
+Author: [Nicholas Chua](https://github.com/nus-cs3281/2018/blob/master/students/nicholasChua/nicholasChua-Resume.md)
+
+## Overview
+
+[ESLint](https://eslint.org/docs/about/) is an open source JavaScript linter.
+
+Code linting is a type of static analysis that helps find problematic patterns or code that does not follow a certain style guideline.
+
+JavaScript, being a loosely-typed, interpreted language, is prone to developer errors and therefore linting is especially important when working with JavaScript.
+
+ESLint is shipped with a wide range of [built-in rules](https://eslint.org/docs/rules/) that can be configured on installation. It also offers flexibility by allowing importing your own custom rules via [plugins](https://eslint.org/docs/developer-guide/working-with-plugins).
+
+ESLint is widely used by many organizations and has [editor integrations](https://eslint.org/docs/user-guide/integrations#editors) such as error highlighting in many popular editors like Atom and WebStorm.
+
+## How it is used
+
+### Installation
+You can [install it with npm](https://eslint.org/docs/user-guide/getting-started#installation-and-usage).
+
+### Configuration
+#### Initial configuration
+ESLint can be configured in one of two ways.
+
+1. Configuration Comments
+1. Configuration File
+
+Both of which are explained [here](https://eslint.org/docs/user-guide/configuring#configuring-rules). You can learn how to set it up [here](https://eslint.org/docs/user-guide/configuring).

```

Comment:
I feel the headings/content ratio is a bit too high. i.e., too many headings. Consider a simpler structure like this:

## Overview

Similar to the current one, but may be structure as 1-2 paragraphs instead of a bunch of separate sentences?

## How it is used

After installing and configuring the rule set, you can run ESLint in the command prompt to see a list of rule violations. It can also be incorporated to your build scripts and CI tools.

* give a list of resources here for installing, configuring, and using ESLint with build/CI tools

## ESLint rules

Give an example rule and describe it.

Mention shared rules from Google etc. here.

## Plugins
...

## Resources
...